tests/MongoDB.Driver.Tests/Specifications/change-streams/ChangeStreamTestRunner.cs

```diff
@@ -112,6 +101,25 @@ private void Run(BsonDocument shared, BsonDocument test)
                 var expectedResult = test["result"].AsBsonDocument;
                 AssertResult(actualResult, actualException, expectedResult);
             }
+
+            BsonArray EmulateRunOn()
+            {
+                var condition = new BsonDocument();
+                if (test.TryGetElement("minServerVersion", out var minServerVersion))
+                {
+                    condition.Add(minServerVersion);
+                }
+                if (test.TryGetElement("maxServerVersion", out var maxServerVersion))
+                {
+                    condition.Add(maxServerVersion);
+                }
+                if (test.TryGetElement("topology", out var topology))
+                {
+                    condition.Add(topology);
+                }
+
+                return new BsonArray().Add(condition);
```

Comment:
nit: we often initialize a `BsonArray` like this:

    return new BsonArray { condition };

but either way is fine.

Reply:
done